Expensify/react-native-live-markdown
# [Mentions v2] Add support for <mention-report /> tag

### Details
This PR is part of [Mentions v2 design doc](https://docs.google.com/document/d/1g_1xLGO29EV6hbOtiYqX53oof_gr8jbnbkSRPjuA4RI/edit). It adds support for new kind of mention: **room mention** which is described with `<mention-report />` HTML tag. This mention should be styled the same as user mention
![image](https://github.com/Expensify/react-native-live-markdown/assets/61146594/3497135d-7d66-4f71-b6de-22d443a4c1e8)


### Related Issues

https://github.com/Expensify/App/issues/39542

### Manual Tests

Type room mention (starts with `#` followed by any combination of small letters, numbers and hyphens ) and verify that
